src/main/java/net/logstash/logback/stacktrace/ShortenedThrowableConverter.java

```diff
@@ -230,16 +248,22 @@ public String convert(ILoggingEvent event) {
         if (throwableProxy == null || isExcludedByEvaluator(event)) {
             return CoreConstants.EMPTY_STRING;
         }
-        
+
+        // compute stack trace hashes
+        Deque<String> stackHashes = null;
```

Comment:
Considering rule 2, we cannot compute error hashes individually.

We could change the `Deque` design, but that would compell to (re)compute several times the root cause errors hash...

Let's discuss about the most appropriate design